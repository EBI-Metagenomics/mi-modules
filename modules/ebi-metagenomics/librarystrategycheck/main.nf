
process LIBRARYSTRATEGYCHECK {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:0.1.5--pyhdfd78af_0':
        'biocontainers/mgnify-pipelines-toolkit:0.1.5--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(mcp_cons)

    output:
    tuple val(meta), path("*.txt"), emit: txt
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    library_strategy_check \\
        -i ${mcp_cons} \\
        -s ${prefix} \\
        -o ./

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: 0.1.5
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}_library_check_out.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: 0.1.5
    END_VERSIONS
    """
}
