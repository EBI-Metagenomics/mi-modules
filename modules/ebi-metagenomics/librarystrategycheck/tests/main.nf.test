
nextflow_process {

    name "Test Process LIBRARYSTRATEGYCHECK"
    script "../main.nf"
    process "LIBRARYSTRATEGYCHECK"

    tag "modules"
    tag "modules_nfcore"
    tag "librarystrategycheck"

    test("library_strategy_check - yes amplicon") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/test_mcp_cons.tsv", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("library_strategy_check - no amplicon") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/test_noamplicon_mcp_cons.tsv", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("library_strategy_check - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/test_mcp_cons.tsv", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
