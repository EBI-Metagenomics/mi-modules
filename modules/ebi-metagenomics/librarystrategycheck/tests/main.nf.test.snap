{
    "library_strategy_check - no amplicon": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_library_check_out.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4921696d5f821a88dfd3e8f362c5862b"
                ],
                "txt": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_library_check_out.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4921696d5f821a88dfd3e8f362c5862b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-08-23T13:58:34.236357618"
    },
    "library_strategy_check - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_library_check_out.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4921696d5f821a88dfd3e8f362c5862b"
                ],
                "txt": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_library_check_out.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4921696d5f821a88dfd3e8f362c5862b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-08-23T14:02:55.745055907"
    },
    "library_strategy_check - yes amplicon": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_library_check_out.txt:md5,1605ca08f9028fe03bcb53a0c50ba2ff"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4921696d5f821a88dfd3e8f362c5862b"
                ],
                "txt": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_library_check_out.txt:md5,1605ca08f9028fe03bcb53a0c50ba2ff"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4921696d5f821a88dfd3e8f362c5862b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-08-23T13:57:28.067289268"
    }
}